src/chemcoord/cartesian_coordinates/_indexers.py

```diff
@@ -10,35 +10,47 @@ class _Loc(_generic_Indexer):
 
     def __getitem__(self, key):
         if isinstance(key, tuple):
-            selected = self.molecule._frame.loc[key[0], key[1]]
+            selected = self.molecule._frame.loc[_set_caster(key[0]), _set_caster(key[1])]
         else:
-            selected = self.molecule._frame.loc[list(key)]
+            selected = self.molecule._frame.loc[_set_caster(key)]
         try:
             return self.molecule._return_appropiate_type(selected)
         except AttributeError:
             return selected
 
     def __setitem__(self, key, value):
         if isinstance(key, tuple):
-            self.molecule._frame.loc[key[0], key[1]] = value
+            self.molecule._frame.loc[_set_caster(key[0]), _set_caster(key[1])] = value
         else:
-            self.molecule._frame.loc[list(key)] = value
+            self.molecule._frame.loc[_set_caster(key)] = value
 
 
 class _ILoc(_generic_Indexer):
 
     def __getitem__(self, key):
         if isinstance(key, tuple):
-            selected = self.molecule._frame.iloc[key[0], key[1]]
+            selected = self.molecule._frame.iloc[_set_caster(key[0]), _set_caster(key[1])]
         else:
-            selected = self.molecule._frame.iloc[list(key)]
+            selected = self.molecule._frame.iloc[_set_caster(key)]
         try:
             return self.molecule._return_appropiate_type(selected)
         except AttributeError:
             return selected
 
     def __setitem__(self, key, value):
         if isinstance(key, tuple):
-            self.molecule._frame.iloc[key[0], key[1]] = value
+            self.molecule._frame.iloc[_set_caster(key[0]), _set_caster(key[1])] = value
         else:
-            self.molecule._frame.iloc[list(key)] = value
\ No newline at end of file
+            self.molecule._frame.iloc[_set_caster(key)] = value
+
+
+
+def _set_caster(x):
```

Comment:
Hello @Mayank447 ,

FYI: This MR is a better way to fix the indexing with sets, since it solves two problems:

1. The unconditional cast to `list` leads to unnecessary errors if something cannot be casted to a list, but is still a valid indexer. The most notable example is a `slice` object, which comes from e.g. `molecule.loc[:]`. Now only sets are casted to lists and for the rest we just pass the object on without casting. 

2. The previous MR did not fix the case of a tuple, i.e. row- and column-indexing, passed to `(i)loc`; i.e.
`loc[{1, 2}, 'x']` did still raise an error because `sets` were not casted in the `if isinstance(key, tuple):` branch.

These problems were not caught in your MR because the CI pipeline did not execute. Now I set up CircleCI again as well.

Reply:
👍